axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -249,8 +250,28 @@ public <R> Registration subscribe(@Nonnull String queryName,
         return new AxonServerRegistration(localRegistration, serverRegistration::cancel);
     }
 
+    private <Q, R> CompletableFuture<QueryResponseMessage<R>> tryToRunQueryLocally(
+            @Nonnull QueryMessage<Q, R> queryMessage) {
+
+        if (localSegment instanceof SimpleQueryBus) {
```

Comment:
When we remove the `public` method you introduced on the `SimpleQueryBus`, we wouldn't be required to perform the `instanceof` check anymore either.